### PR TITLE
NMS-15008: Replace community string in poller requests

### PR DESCRIPTION
**Pull request by amay94**
### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-15008

### CODE REVIEW

**Review by fooker — COMMENTED** [review]
Why do this in the RPC module? Could you just override/add an attribute using the runtime-attributes mechanism?

**Review by RangerRick — DISMISSED** [review]
I'm not going to pretend to understand the intricacies of SCV, but this seems pretty straightforward, and seems pretty well-bounded to not accidentally blow up.

File: features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/SnmpMonitor.java [hunk 1]
```diff
@@ -107,6 +111,30 @@ public PollStatus poll(MonitoredService svc, Map<String, Object> parameters) {
         String reasonTemplate = ParameterMap.getKeyedString(parameters, "reason-template", DEFAULT_REASON_TEMPLATE);
         String hexstr = ParameterMap.getKeyedString(parameters, "hex", "false");
 
+        String location = ParameterMap.getKeyedString(parameters, "location", LocationUtils.DEFAULT_LOCATION_NAME);
```
Comment (cgorantla) [inline features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/SnmpMonitor.java:114]: I think you can get location from `scv.getNodeLocation()`

**Review by aramosvizcarra — DISMISSED** [review]
They seem to do the job and the code looks good from my end

File: features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/SnmpMonitor.java [hunk 2]
```diff
@@ -107,6 +111,30 @@ public PollStatus poll(MonitoredService svc, Map<String, Object> parameters) {
...
+
+        if (!LocationUtils.DEFAULT_LOCATION_NAME.equals(location) && agentConfig.getAddress().isLoopbackAddress()) {
```
Comment (cgorantla) [inline features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/SnmpMonitor.java:116]: Another way to access scv by overriding  `@Override
    public Map<String, Object> getRuntimeAttributes(MonitoredService svc, Map<String, Object> parameters)`. and get read community there and send it as a parameter.

**Review by cgorantla — COMMENTED** [review]
LGTM, few comments.

File: features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/SnmpMonitor.java (hunk 2 above)
Comment (cgorantla) [reply features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/SnmpMonitor.java]: As @amay94 pointed, we can't use `getRuntimeAttributes` for this as the scv needs to be retrieved only on Minion and `getRuntimeAttributes` only runs on OpenNMS

**Review by cgorantla — APPROVED** [review]
LGTM!

**Pull request merged**